src/modules/interface/controller_indi.h

```diff
@@ -0,0 +1,125 @@
+/**
+ *    ||          ____  _ __
+ * +------+      / __ )(_) /_______________ _____  ___
+ * | 0xBC |     / __  / / __/ ___/ ___/ __ `/_  / / _ \
+ * +------+    / /_/ / / /_/ /__/ /  / /_/ / / /_/  __/
+ *  ||  ||    /_____/_/\__/\___/_/   \__,_/ /___/\___/
+ *
+ * Crazyflie control firmware
+ *
+ * Copyright (C) 2011-2016 Bitcraze AB
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, in version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * controller_indi.h - INDI Controller Interface
+ */
+#ifndef __CONTROLLER_INDI_H__
+#define __CONTROLLER_INDI_H__
+
+#include "stabilizer_types.h"
+#include "filter.h"
+#include "math3d.h"
+
+#define PI 3.14159265358979323846f
+
+//The G values are scaled to avoid numerical problems during the estimation
+#define INDI_EST_SCALE 0.001f
+
+// these parameters are used in the filtering of the angular acceleration
+#define STABILIZATION_INDI_FILT_CUTOFF 8.0f
+
+// the yaw sometimes requires more filtering
+#define STABILIZATION_INDI_FILT_CUTOFF_R STABILIZATION_INDI_FILT_CUTOFF
+
+#define STABILIZATION_INDI_ESTIMATION_FILT_CUTOFF 4.0f
+
+#define PERIODIC_FREQUENCY 512.0f
```

Comment:
The frequency of the controller is specified somewhere else in the Crazyflie code, please use that value

Reply:
As far as I understood, we should instead be using the `ATTITUDE_RATE` specified in the `stabilizer_types.h`, which corresponds to the IMU frequency. Is that correct?

Reply:
I think that is correct. Can you take a look what the function of this line is?
https://github.com/amreelab/crazyflie-firmware/blob/indi-controller/src/modules/src/controller_mellinger.c#L133

Reply:
It seems to be testing the `ATTITUDE_RATE` (500Hz) in relation to the `MAIN_LOOP_RATE` (1000Hz), trying to guarantee that the `TICK` is a multiple of this relation in order to keep running the controller. Do you understand what is the reason of this test? Should we be doing this test too?

Reply:
https://github.com/amreelab/crazyflie-firmware/blob/indi-controller/src/modules/src/stabilizer.c#L210

It think that answers your question 😉

Reply:
Inserted this verification to skip calls faster than ATTITUDE_RATE.